### PR TITLE
Add a basic Au test using strategies.yaml, with consistent hashing.

### PR DESCRIPTION
**Pull request by ywkaras**

### CODE REVIEW

**Comment by ywkaras** [comment]
[approve ci clang-format]

**Comment by ywkaras** [comment]
[approve ci autest]

File: tests/gold_tests/next_hop/strategies_ch/strategies_ch.test.py [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+s.AddLine("    groups:")
+s.AddLine("      - *g1")
+s.AddLine("    scheme: http")
+#s.AddLine("    fallover:")
+#s.AddLine("      max_simple_retries: 2")
+#s.AddLine("      ring_mode: exhaust_ring")
+#s.AddLine("      response_codes:")
+#s.AddLine("        - 404")
+#s.AddLine("      health_check:")
+#s.AddLine("        - passive")
```
Comment (bneradt) [inline tests/gold_tests/next_hop/strategies_ch/strategies_ch.test.py:96]: There's three places where there is commented out code in this test. We should probably either remove them or add comments above them explaining why they are being left in commented out (for instance, if they can be useful for debugging if something's going wrong).

File: tests/gold_tests/next_hop/strategies_ch/strategies_ch.test.py [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    s.AddLine(f"    - host: next_hop{i}")
+    s.AddLine(f"      protocol:")
+    s.AddLine(f"        - scheme: http")
+    s.AddLine(f"          port: {server_nh[i].Variables.Port}")
+    #s.AddLine(f"          health_check_url: http://next_hop{i}:{server_nh[i].Variables.Port}")
+    s.AddLine(f"      weight: 1.0")
+s.AddLine("")
+s.AddLine("strategies:")
+s.AddLine("  - strategy: the-strategy")
+s.AddLine("    policy: consistent_hash")
```
Comment (bneradt) [inline tests/gold_tests/next_hop/strategies_ch/strategies_ch.test.py:82]: Can `AddLines` be used more naturally for this?

**Review by bneradt — CHANGES_REQUESTED** [review]
Looks good. I just have a few small suggestions.

File: tests/gold_tests/next_hop/strategies_ch/strategies_ch.test.py (hunk 1 above)
Comment (ywkaras) [reply tests/gold_tests/next_hop/strategies_ch/strategies_ch.test.py:96]: It's simply there to make it easier to try non-default configurations.

File: tests/gold_tests/next_hop/strategies_ch/strategies_ch.test.py (hunk 2 above)
Comment (ywkaras) [reply tests/gold_tests/next_hop/strategies_ch/strategies_ch.test.py:82]: Is this documented anywhere?

Comment (ywkaras) [reply tests/gold_tests/next_hop/strategies_ch/strategies_ch.test.py:82]: I guess I can do the usual monkey see monkey do.

Comment (bneradt) [reply tests/gold_tests/next_hop/strategies_ch/strategies_ch.test.py:82]: Looking into this, the AddLine and AddLines functions are actually not AuTest framework functions, but extension functions added in the autest-site directory:

https://github.com/apache/trafficserver/blob/01d2150ef43650fec904c9ddd0d85978cd0a7390/tests/gold_tests/autest-site/trafficserver.test.ext#L396

**Comment by ywkaras** [comment]
@bneradt this is ready for re-review.

**Comment by ywkaras** [comment]
Not sure this should be merged.  A similar, but more complex test (also tests fallover), has already been merged:  https://github.com/apache/trafficserver/pull/7914

**Review by bneradt — APPROVED** [review]

**Review by jrushford — APPROVED** [review]
looks good

**Comment by zwoop** [comment]
Cherry-picked to v9.1.x branch.

**Pull request merged**